camel-idea-plugin/src/main/java/org/apache/camel/idea/annotator/CamelAnnotator.java

```diff
@@ -0,0 +1,220 @@
+        if ((fromElement.contains("?") || fromElement.contains("&")) && IdeaUtils.isConsumerEndpoint(element)) {
+            CamelCatalog catalogService = ServiceManager.getService(element.getProject(), CamelCatalogService.class).get();
+            EndpointValidationResult validateEndpointProperties = catalogService.validateEndpointProperties(fromElement);
+            extractMapValue(validateEndpointProperties, validateEndpointProperties.getInvalidBoolean(), fromElement, element, holder);
+            extractMapValue(validateEndpointProperties, validateEndpointProperties.getInvalidEnum(), fromElement, element, holder);
+            extractMapValue(validateEndpointProperties, validateEndpointProperties.getInvalidInteger(), fromElement, element, holder);
+            extractMapValue(validateEndpointProperties, validateEndpointProperties.getInvalidNumber(), fromElement, element, holder);
+            extractMapValue(validateEndpointProperties, validateEndpointProperties.getInvalidReference(), fromElement, element, holder);
+        }
+    }
+
+    // TODO : Solve classloader issue class not found when CamelCatalog access a camel core java file.
+    // The problem is releated to the CamelCatalog is using the plugin classloader, but the Camel core
+    // exits on the project/module classloader
+    /**
+     * Validate simple expression. eg "simple("${body}"
+     * if the expression is not valid a error annotation is created and highlight the invalid value.
+     */
+    private void validateSimpleExpression(@NotNull PsiElement element, @NotNull AnnotationHolder holder, String fromElement) {
+        if (fromElement.contains("${") &&  (IdeaUtils.isCamelRouteSimpleExpression(element))) {
+            CamelCatalog catalogService = ServiceManager.getService(element.getProject(), CamelCatalogService.class).get();
+            SimpleValidationResult validateSimpleResult = catalogService.validateSimpleExpression(fromElement);
+            if (!validateSimpleResult.isSuccess()) {
+                String error = validateSimpleResult.getError();
+                int propertyIdx = fromElement.indexOf("simple");
+                int propertyLength = fromElement.length() + 8;
+                TextRange range = new TextRange(element.getTextRange().getStartOffset() + propertyIdx,
+                    element.getTextRange().getStartOffset() + propertyIdx + propertyLength);
+                holder.createErrorAnnotation(range, error);
+            }
+
+        }
+    }
+
+    private void extractMapValue(EndpointValidationResult validateEndpointProperties, Map<String, String> validationMap,
+                                 String fromElement, @NotNull PsiElement element, @NotNull AnnotationHolder holder) {
+        if ((!validateEndpointProperties.isSuccess()) && validationMap != null) {
+            String propertyValue = validationMap.values().iterator().next();
```

Comment:
It actually look like you can create multiple annotations with different text range for the same AnnotationHolder. It was not clear in the API but after I look through the code I can see it add a new annotator to a list behind the scene